src/lobster/model/_ume.py

```diff
@@ -142,11 +142,9 @@ def __init__(
         self.embedding_dim = self.model.config.hidden_size
         self.frozen = False
 
-        self.metrics = {}
-
         for modality in Modality:
-            self.metrics.update({f"train_perplexity/{modality.value}": Perplexity(ignore_index=-100)})
-            self.metrics.update({f"val_perplexity/{modality.value}": Perplexity(ignore_index=-100)})
+            setattr(self, f"train_perplexity/{modality.value}", Perplexity(ignore_index=-100))
```

Comment:
can you add a comment here about the lightning behavior and need for `setattr` ?